lib/FrontendTool/FrontendTool.cpp

```diff
@@ -353,15 +353,16 @@ class ABIDependencyEvaluator {
   ///
   /// \code
   /// Actual stack:
-  ///    sandwichedModule -> Overlay (Swift) -> ... -> sandwichedModule
-  ///                     ^^--- wrong!
+  ///    sandwichedModule -> .. -> Overlay (Swift) -> .. -> sandwichedModule
+  ///                           ^^--- wrong!
   /// Ideal stack:
-  ///    sandwichedModule -> Underlying (Clang)
+  ///    sandwichedModule -> .. -> Underlying (Clang)
   /// \endcode
+  /// (The first series of \c .. values are Clang modules.)
   ///
   /// This happens when we have a dependency like:
   /// \code
-  ///     Overlay (Swift) -> sandwichedModule -> Underlying (Clang)
+  ///     Overlay (Swift) -> sandwichedModule -> .. -> Underlying (Clang)
```

Comment:
@varungandhi-apple With both more distance from the original PR and the description of the cycles it detects becoming more abstract, I find I'm not really clear about what `isFakeCycleThroughOverlay()` should do. I think this is down to the doc comment requiring too much context to understand. It ought to describe:

* When these cycles occur
* What should be done about them (I believe this is "ignore them because they're benign", but that's not clear from the comment)
* What the relationship should be between the various modules involved in them (e.g. is `Overlay` always the overlay of `Underlying`? How is `sandwichedModule` related to them? How are the other modules in between related to them?)

I think I'll have more success reviewing the actual implementation once I can compare it to a clearer description of what it *ought* to do.

Reply:
I've tried to explain things with a concrete example and point out what can happen in the most general case, instead of focusing on the most general thing in the comment.